src/IW/Sync/Search.hs

```diff
@@ -0,0 +1,89 @@
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchHaskellIssuesByLabels
+       , githubIssueToIssue
+       , githubRepoToRepo
+       ) where
+
+import Data.Text (split)
+import GitHub (SearchResult (..))        
+import GitHub.Endpoints.Search (searchRepos, searchIssues)
+
+import IW.App (WithError)
+import IW.Core.Issue (Issue (..), Label (..))
+import IW.Core.Repo (Repo (..), RepoOwner (..), RepoName (..))
+import IW.Core.SqlArray (SqlArray (..))
+import IW.Db (WithDb)
+import IW.App.Error (githubErrToAppErr, throwError)
+
+import qualified GitHub
+import qualified Data.Vector as Vector
+
+
+-- | Fetch all repositories built with the Haskell language.
+fetchAllHaskellRepos :: (WithDb env m, WithError m, WithLog env m) => m [GitHub.Repo]
+fetchAllHaskellRepos = liftGitHubSearchToApp searchHaskellRepos 
+
+-- | Fetch all open issues with Haskell language and the labels passed in to the function.
+fetchHaskellIssuesByLabels :: (WithDb env m, WithError m, WithLog env m) => [Label] -> m [GitHub.Issue]
+fetchHaskellIssuesByLabels labels = liftGitHubSearchToApp $ searchHaskellIssuesByLabels labels
+
+-- | Fetch all open issues with Haskell language.
+fetchAllHaskellIssues :: (WithDb env m, WithError m, WithLog env m) => m [GitHub.Issue]
+fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
+
+-- | Lift a github search action to work within our monad stack.
+liftGitHubSearchToApp :: (WithDb env m, WithError m, WithLog env m) => IO (Either GitHub.Error (SearchResult a)) -> m [a]
+liftGitHubSearchToApp githubSearch = do
+    result <- liftIO githubSearch
+    case result of
+        Right (SearchResult count vec) -> do
+            log Info $ "Fetching total of " <> show count <> " results." 
```

Comment:
You can add `Typeable`  constraint to `a` type variable and use `typeName @a` here instead of `"results"` to log what data type we are processing now.

Reply:
Okay cool I was wondering how to do that. Thank you.

Reply:
```
/home/rashad/Desktop/issue-wanted/src/IW/Sync/Search.hs:66:63: error:
    • Cannot apply expression of type ‘t1’
      to a visible type argument ‘a’
    • In the first argument of ‘(<>)’, namely ‘(typeName @a)’
      In the second argument of ‘(<>)’, namely ‘(typeName @a) <> "s..."’
      In the second argument of ‘(<>)’, namely
        ‘show count <> (typeName @a) <> "s..."’
   |        
66 |             log Info $ "Fetching total of " <> show count <> (typeName @a) <> "s..." 
   |                                                               ^^^^^^^^^^^
            
Progress 1/2
```
I'm getting this error when doing `typeName @a`. Can't figure out why. Am I missing a language extension? I tried adding `ExplicitForAll` but nothing happened.

Reply:
You need to add `forall` explicitly to the function signature. The extension is `ScopedTypeVariables`. It's enabled by default in this project.